pontoon/base/views.py

```diff
@@ -988,50 +994,119 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def download_file(prefixes, dirnames, relative_path):
+    for prefix in prefixes:
+        for dirname in dirnames:
+            url = os.path.join(prefix.replace('{locale_code}', dirname), relative_path)
+            r = requests.get(url, stream=True)
+            if not r.ok:
+                continue
+
+            filename = str(uuid.uuid4()) + os.path.splitext(url)[1]
```

Comment:
Yes, because

1. They're guaranteed to be made in a temporary directory that won't affect other programs incidentally.
2. As jotes mentioned, if your code fails to delete the file due to an error, it will be removed later by the OS (at least on Linux and OSX).
3. tempfile makes some guarantees about no race conditions during creation that are desireable.

But really to me, it's that, when given two roughly equivalent options with the same outcome, choose the semantically correct one. It's more correct to use the stdlib's functions for making temporary files rather than using other functions that accomplish the same thing but aren't for making temporary files.